ledger-api/grpc-definitions/com/daml/ledger/api/v1/error.proto

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v1;
+
+// Rules on error reporting over the Ledger API
+//
+// 1. We use gRPC status codes to report synchronous success and failures
+//    coming from the underlying gRPC protocol and at the API level.
+//    Asynchronous API-level errors are reported over a completion stream like the command completion stream,
+//    which themselves include a `google.rpc.Status` message to indicate the status.
+//    The status code `OK` denotes success.
+//
+// 2. Error reporting includes relevant error details in the `details` field of the status message,
+//    following the gRPC rich error model as described at https://cloud.google.com/apis/design/errors#error_model.
+//    Error details use the `google.rpc.ErrorInfo` message defined at
+//    https://github.com/googleapis/googleapis/blob/master/google/rpc/error_details.proto#L112.
```

Comment:
@andreaslochbihler-da : is the plan to adopt an error payloads table as in https://cloud.google.com/apis/design/errors#error_payloads, but for now we only use `ErrorInfo`?

Reply:
Do you have plans to adopt https://github.com/googleapis/googleapis/blob/599fed858da3a3fdd59a2530046ee506928da25b/google/rpc/error_details.proto#L40 for the cases where clients can retry?

It might though not be sufficiently expressive for our retry cases.

Reply:
1. For now, I'd go with `ErrorInfo` only. I find it more important to document somehow what error details we provide rather than creating sub-divisions like in the linked table. We can create such a table once we've built up some experience with our API.

2. `RetryInfo` is worth looking at when we actually deal with plain retryable errors. For command dedup, only few error situations are "retry as is", so I don't think it is worth to think about this for the command dedup work.